File: java/1310_XOR_Queries_of_a_Subarray.java
class Solution {
    public int[] xorQueries(int[] arr, int[][] queries) {
        int narr = arr.length;
        int nqueries = queries.length;
        int[] sum = new int[narr + 1];

        for (int i = 1 ; i <= narr ; i++){
            sum[i] = sum[i-1] ^ arr[i-1];
        }

        int[] ans = new int[nqueries];
        for(int i=0; i < nqueries ; i++){
            ans[i] = sum[queries[i][0]] ^ sum[queries[i][1]+1];
        }
        return ans;
    }
}

File: python/1310_XOR_Queries_of_a_Subarray.py
class Solution:
    def countTriplets(self, arr: List[int]) -> int:
        res = 0;
        for i in range(len(arr)):
            t = arr[i]
            for k in range(i+1,len(arr),1):
                t ^= arr[k]
                if t == 0:
                    res += k - i
        return res